assets/js/programmingLanguageSwitch.js

```diff
@@ -0,0 +1,61 @@
+// Set the language preference in localStorage and update the displayed content immediately
+function setLanguagePreference(language, pagePath) {
+    // Set language preference in localStorage
+    localStorage.setItem('languagePreference_' + pagePath, language);
+    // Update the displayed content based on the selected language
+    showContent(language, pagePath);
+}
+
+// Get the language preference from localStorage
+function getLanguagePreference(pagePath) {
+    return localStorage.getItem('languagePreference_' + pagePath);
+}
+
+// Show the content based on the selected language
+function showContent(language, pagePath) {
+    // Map each language to its corresponding content element
+    var contentMap = {
+        'cpp': 'cppContent',
+        'csharp': 'csharpContent',
+        'python': 'pythonContent'
```

Comment:
Will we need to add each new language here as we expand?

Reply:
Yes, the contentMap  " names " the second part of it is what you would reference in the " <div> " of the .md files and only ones written on the contentMap can be used, I didn't find a better work around for it as of currently, but it should work for the current needs.  Also not that you will only add the <div>   and buttons depending what you need.

Reply:
Sure, seems fair